source/dem/insertion.cc

```diff
@@ -197,5 +221,131 @@ Insertion<dim>::calculate_insertion_domain_maximum_particle_number(
       inserted_this_step = insertion_information.inserted_this_step;
     }
 }
+
+
+template <int dim>
+void
+Insertion<dim>::find_in_clearing_box_cells(
+  const parallel::distributed::Triangulation<dim> &triangulation)
+{
+  // Clearing the containers
+  in_the_clearing_box.clear();
+  edge_of_box.clear();
+  bool partially_inside, completely_inside;
+  // Looping through cells
+  for (const auto &cell : triangulation.active_cell_iterators())
+    {
+      if (cell->is_locally_owned())
+        {
+          partially_inside  = false;
+          completely_inside = true;
+          // Loop through its vertices
+          for (unsigned int vertex_id = 0; vertex_id < cell->n_vertices();
+               ++vertex_id)
+            {
+              Point<3> vertex = point_nd_to_3d(cell->vertex(vertex_id));
+
+              // Check if the n_th vertex is in the box
+              if (p_min[0] <= vertex[0] && vertex[0] <= p_max[0] &&
+                  p_min[1] <= vertex[1] && vertex[1] <= p_max[1] &&
+                  p_min[2] <= vertex[2] && vertex[2] <= p_max[2])
```

Comment:
It is cleaner if you just loop and store it in a variable, but like this it is fine

Reply:
Or even, if you make an is_within_bounds member parsing the containers as parameters, this should be even cleaner and useful for other stuff.